Change gcc build : Enable m4 and m4-nofpu (needed for kernel) for sh4-gnu-linux

Signed-off-by: Carl Shaw <carl.shaw@st.com>

diff -Naur gcc-4.2.1-compteam/gcc/config.gcc gcc-4.2.1-stm/gcc/config.gcc
--- gcc-4.2.1-compteam/gcc/config.gcc	2007-08-10 13:08:59.000000000 +0100
+++ gcc-4.2.1-stm/gcc/config.gcc	2007-08-14 10:37:37.000000000 +0100
@@ -2094,11 +2094,12 @@
 	if test x${sh_multilibs} = x ; then
 		case ${target} in
 		sh64-superh-linux* | \
-		sh[1234]*)	sh_multilibs=${sh_cpu_target} ;;
 		sh64* | sh5*)	sh_multilibs=m5-32media,m5-32media-nofpu,m5-compact,m5-compact-nofpu,m5-64media,m5-64media-nofpu ;;
-		sh-superh-*)	sh_multilibs=m4,m4-single,m4-single-only,m4-nofpu ;;
+		sh-superh-* | \
+		sh4-*-linux*)	sh_multilibs=m4,m4-nofpu ;;
 		sh*-*-linux*)	sh_multilibs=m1,m3e,m4 ;;
 		sh*-*-netbsd*)	sh_multilibs=m3,m3e,m4 ;;
+		sh[1234]*)	sh_multilibs=${sh_cpu_target} ;;
 		*) sh_multilibs=m1,m2,m2e,m4,m4-single,m4-single-only,m2a,m2a-single ;;
 		esac
 		if test x$with_fp = xno; then
